highfidelity/hifi_tests
# Convert protocol test results to text

# Description
A text file is produced for the protocol tests.  This text file is in addition to the green/red (pass/fail images).  In a future version of nitpick, these images will  be removed and nitpick will check the text files.

Attached is an example of a test that passed.
[tests.protocol.box.passed.txt](https://github.com/highfidelity/hifi_tests/files/2630111/tests.protocol.box.passed.txt)
# Testing
1.  Start interface from the command line (build number is not important): `interface.exe --url "hifi://localhost" --no-login-suggestion --no-updater --testScript https://raw.githubusercontent.com/NissimHadar/hifi_tests/convertProtocolTestResultsToText/tests/protocol/testRecursive.js quitWhenFinished --testResultsLocation xxxx`, where xxxx is an empty folder.
3.  When the script has completed, verify that the snapshots folder has 3 files for each of 7 tests.  The files for each test consist of 2 images and a text file (there will be an additional test file named `tests_completed.txt`).
4.  Verify that all images ending in `00001.png` show a green box.
5.  Verify that each of the 7 relevant  text files ends  in `.passed.txt`.
